fastly/block_fastly_service_logging_newrelicotlp.go

```diff
@@ -0,0 +1,264 @@
+	}
+	if v, ok := modified["url"]; ok {
+		opts.URL = gofastly.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Update New Relic OTLP Opts: %#v", opts)
+	_, err := conn.UpdateNewRelicOTLP(&opts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Delete deletes the resource.
+func (h *NewRelicOTLPServiceAttributeHandler) Delete(_ context.Context, d *schema.ResourceData, resource map[string]any, serviceVersion int, conn *gofastly.Client) error {
+	opts := h.buildDelete(resource, d.Id(), serviceVersion)
+
+	log.Printf("[DEBUG] Fastly New Relic OTLP logging endpoint removal opts: %#v", opts)
+
+	return deleteNewRelicOTLP(conn, opts)
+}
+
+func createNewRelicOTLP(conn *gofastly.Client, i *gofastly.CreateNewRelicOTLPInput) error {
+	_, err := conn.CreateNewRelicOTLP(i)
+	return err
+}
+
+func deleteNewRelicOTLP(conn *gofastly.Client, i *gofastly.DeleteNewRelicOTLPInput) error {
+	err := conn.DeleteNewRelicOTLP(i)
+
+	errRes, ok := err.(*gofastly.HTTPError)
+	if !ok {
+		return err
+	}
+	// 404 response codes don't result in an error propagating because a 404 could
+	// indicate that a resource was deleted elsewhere.
+	if !errRes.IsNotFound() {
+		return err
+	}
+	return nil
+}
+
+// flattenNewRelicOTLP models data into format suitable for saving to Terraform state.
+func flattenNewRelicOTLP(remoteState []*gofastly.NewRelicOTLP) []map[string]any {
+	var result []map[string]any
+	for _, resource := range remoteState {
+		data := map[string]any{
+			"name":               resource.Name,
+			"token":              resource.Token,
+			"format":             resource.Format,
+			"format_version":     resource.FormatVersion,
+			"placement":          resource.Placement,
+			"region":             resource.Region,
+			"response_condition": resource.ResponseCondition,
+			"url":                resource.URL,
+		}
+
+		// Prune any empty values that come from the default string value in structs.
+		for k, v := range data {
+			if v == "" {
+				delete(data, k)
+			}
+		}
+
+		result = append(result, data)
+	}
+
+	return result
+}
+
+func (h *NewRelicOTLPServiceAttributeHandler) buildCreate(newrelicotlpMap any, serviceID string, serviceVersion int) *gofastly.CreateNewRelicOTLPInput {
+	resource := newrelicotlpMap.(map[string]any)
+
+	vla := h.getVCLLoggingAttributes(resource)
+	opts := &gofastly.CreateNewRelicOTLPInput{
+		Format:         gofastly.String(vla.format),
+		FormatVersion:  vla.formatVersion,
+		Name:           gofastly.String(resource["name"].(string)),
+		Region:         gofastly.String(resource["region"].(string)),
+		ServiceID:      serviceID,
+		ServiceVersion: serviceVersion,
+		Token:          gofastly.String(resource["token"].(string)),
+		URL:            gofastly.String(resource["url"].(string)),
+	}
+
+	// WARNING: The following fields shouldn't have an empty string passed.
+	// As it will cause the Fastly API to return an error.
+	// This is because go-fastly v7+ will not 'omitempty' due to pointer type.
```

Comment:
Is this definitely the case for this API endpoint? I just want to be sure we're not copy/pasting (aka cargo-culting) a pattern that might be relevant only for some other endpoints.

Reply:
Yes, I believe this is true. I made a quick test program using gofastly and tried creating an endpoint with empty strings for both `ResponseCondition` and `Placement` and got 400 errors for each

```
error: 400 - Bad Request:

    Title:  Bad request
    Detail: No condition named ''
    RateLimitRemaining: 994
    RateLimitReset:     1702666800
```

```
error: 400 - Bad Request:

    Title:  Bad request
    Detail: Not allowed placement for ''
    RateLimitRemaining: 993
    RateLimitReset:     1702666800

```